R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,447 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
+      filter(year %in% FUTURE_YEARS) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) %>%
+      mutate(storage.market = "carbon-storage") ->
+      L2322.GlobalTechCapture_Fert
+
+    # Retirement information
+    A322.globaltech_retirement %>%
+      set_years %>%
+      mutate(year = as.numeric(year)) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      A322.globaltech_retirement_with_years
+
+    # Copy the data in the first future period through to the end year
+    A322.globaltech_retirement_with_years %>%
+      filter(year == max(BASE_YEARS)) ->
+      A322.globaltech_retirement_max_baseyear
+
+    A322.globaltech_retirement_with_years %>%
+      filter(year == min(FUTURE_YEARS)) %>%
+      select(-year) %>%
+      repeat_add_columns(tibble(year = FUTURE_YEARS)) %>%
+      bind_rows(A322.globaltech_retirement_max_baseyear) ->
+      L2322.globaltech_retirement
+
+    # Retirement may consist of any of three types of retirement function (phased, s-curve, or none)
+    # All of these options have different headers, and all are allowed
+
+    # L2322.GlobalTechShutdown_Fert: Global tech lifetime and shutdown rate
+    tibble(x = NA) %>%
+      add_units("None") %>%
+      add_comments("Not generated") %>%
+      add_flags(FLAG_NO_TEST) ->
+      L2322.GlobalTechShutdown_Fert
+
+    # L2322.GlobalTechSCurve_Fert: Global tech lifetime and s-curve retirement function
+    L2322.globaltech_retirement %>%
+      filter(!is.na(half.life)) %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["GlobalTechYr"]], "lifetime", "steepness", "half.life"))) ->
+      L2322.GlobalTechSCurve_Fert
+
+    # L2322.GlobalTechLifetime_Fert: Global tech lifetime
+    tibble(x = NA) %>%
+      add_units("None") %>%
+      add_comments("Not generated") %>%
+      add_flags(FLAG_NO_TEST) ->
+      L2322.GlobalTechLifetime_Fert
+
+    # L2322.GlobalTechProfitShutdown_Fert: Global tech profit shutdown decider.
+    L2322.globaltech_retirement %>%
+      filter(!is.na(median.shutdown.point)) %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["GlobalTechYr"]], "median.shutdown.point", "profit.shutdown.steepness"))) ->
+      L2322.GlobalTechProfitShutdown_Fert
+
+    # Calibration and region-specific data
+    # L2322.StubTechProd_Fert: calibrated output of fertilizer technologies
+    L1322.Fert_Prod_MtN_R_F_Y %>%
+      filter(year %in% BASE_YEARS) %>%
+      rename(calOutputValue = value) %>%
+      mutate(calOutputValue = round(calOutputValue, energy.DIGITS_CALOUTPUT)) %>%
+      left_join_error_no_match(GCAM_region_names, by = 'GCAM_region_ID') %>%
+      left_join_error_no_match(select(calibrated_techs, sector, fuel, supplysector, subsector, technology), by = c("sector", "fuel") ) %>%
+      rename(stub.technology = technology) %>%
+      mutate(share.weight.year = year, subs.share.weight = 0, subs.share.weight = replace(subs.share.weight, which(calOutputValue > 0), 1), tech.share.weight = subs.share.weight) %>%
```

Comment:
Wouldn't it be easier to do an `if_else` on `subs.share.weight.`

Reply:
Thanks! I had a false impression that `if_else` is forbidden. Turned out only `ifelse` cannot be used.